examples/lines_bars_and_markers/scatter_hist.py

```diff
@@ -3,23 +3,63 @@
 Scatter Hist
 ============
 
-Creates histogram from scatter plot
-and adds them to the sides of the plot.
+Show the marginal distributions of a scatter as histograms at the sides of
+the plot.
+
+For a nice alignment of the main axes with the marginals, two options are shown
+below.
+
+* the axes positions are defined in terms of rectangles in figure coordinates
+* the axes positions are defined via a gridspec
+
+An alternative method to produce a similar figure using the ``axes_grid1``
+toolkit is shown in the
+:doc:`/gallery/axes_grid1/scatter_hist_locatable_axes` example.
 
 """
+
+#############################################################################
+# Let us first define a function that takes x and y data as input, as well
```

Comment:
There will be no space in the generated ReST code between the docstring (stripped) and the documentation comment. Here, the two texts get merged into one paragraph.

Not sure if you can do this at all or if you have to continue the docstring. Anyway this should have a section title.

Reply:
That is actually desired - in the sense of, I do not want a new section here (because the two sections are the two examples mentionned earlier). I now put everything in the heading docstring.